frontend/public/components/pod.tsx

```diff
@@ -911,28 +1004,55 @@ export const PodsPage = connect<{}, PodPagePropsFromDispatch, PodPageProps>(
         }
       }, [namespace]);
       /* eslint-enable react-hooks/exhaustive-deps */
+
+      const [pods, loaded, loadError] = useK8sWatchResource<PodKind[]>({
+        kind,
+        isList: true,
+        namespaced: true,
+        namespace,
+        selector,
+        fieldSelector,
+      });
+
+      const [data, filteredData, onFilterChange] = useListPageFilter(pods);
+
+      const filters = React.useMemo(() => getFilters(t), [t]);
+
       return (
-        <ListPage
-          {...listProps}
-          canCreate={canCreate}
-          kind={kind}
-          ListComponent={PodList}
-          rowFilters={getFilters()}
-          namespace={namespace}
-          customData={customData}
-          columnLayout={{
-            columns: getHeader(props?.customData?.showNodes)().map((column) =>
-              _.pick(column, ['title', 'additional', 'id']),
-            ),
-            id: columnManagementID,
-            selectedColumns:
-              tableColumns?.[columnManagementID]?.length > 0
-                ? new Set(tableColumns[columnManagementID])
-                : null,
-            showNamespaceOverride: props?.customData?.showNamespaceOverride,
-            type: t('public~Pod'),
-          }}
-        />
+        <>
+          <ListPageHeader title={showTitle ? t('public~Pods') : undefined}>
```

Comment:
I think this is a better API in general. It's more flexible and composable. It's a bit more code for the resource pages compared to using `ListPage`, but I think that's a reasonable tradeoff and the common header/filter/list components will help keep consistency. This will also allow dynamic plugins to do something special when needed without having to change `ListPage`.